File: problems/08_tries/0208_implement_trie_prefix_tree/solution.swift
// Problem: 0208. Implement Trie (Prefix Tree)
// Difficulty: Medium

import Foundation

final class TrieNode {
    var ch: [Character: TrieNode] = [:]
    var end: Bool = false
}

class Trie {
    private let root = TrieNode()
    init() {
        // TODO: implement
    }
    func insert(_ word: String) {
        // TODO: implement
    }
    private func walk(_ s: String) -> TrieNode? {
        // TODO: implement
        return nil
    }
    func search(_ w: String) -> Bool {
        // TODO: implement
        return false
    }
    func startsWith(_ p: String) -> Bool {
        // TODO: implement
        return false
    }
}

func checkBool(_ a: Bool, _ e: Bool, _ msg: String) {
    if a != e { fatalError("FAIL \(msg)") }
}

// MARK: - Tests
// let s = Solution()

// Test 1
// let result = s.solve(...)

print("All tests passed!")

File: problems/08_tries/0208_implement_trie_prefix_tree/solution.dart
// Problem: 0208. Implement Trie (Prefix Tree)
// Difficulty: Medium

class _Node {
  final Map<String, _Node> ch = {};
  bool end = false;
}

class Trie {
  final _Node root = _Node();
  void insert(String word) {
    // TODO: implement
  }
  _Node? _walk(String s) {
    // TODO: implement
    return null;
  }
  bool search(String w) {
    // TODO: implement
    return false;
  }
  bool startsWith(String p) {
    // TODO: implement
    return false;
  }
}


void main() {
  // Test 1
  // final result = solve(...);
  print('All tests passed!');
}
